### ISSUE TITLE
Since 3.1, calling a Java keyword from a Python keyword raises TypeError

### ISSUE THREAD

**Issue by Soothsilver** [robotframework/robotframework#3036]
In 3.1, if I use a Java keyword by calling BuiltIn().run_keyword() from within a Python keyword (and the Java keyword writes on standard output), the Python keyword will run until its end correctly but then raise this error:

```
TypeError: 'NoneType' object is not iterable
Traceback (most recent call last):
  File "C:\src\git\shared\robots\itas-robot\target\lib\robotframework-3.1.jar\Lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\src\git\shared\robots\itas-robot\target\lib\robotframework-3.1.jar\Lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\src\git\shared\robots\itas-robot\target\lib\robotframework-3.1.jar\Lib\robot\output\logger$py.class", line 171, in delayed_logging
```
With 3.0.4, this works okay.

**Comment by pekkaklarck** [comment]
Do you use timeouts? If yes, then this is most likely a duplicate of #3025. If not, then the root cause and hopefully the fix is most likely the same. Could you test does the error go away with the recently released RF 3.1.1 rc 1?

https://github.com/robotframework/robotframework/blob/master/doc/releasenotes/rf-3.1.1.rst

**Comment by Soothsilver** [comment]
I use timeout and it appears to be the same issue. I'm using the Java/Jython packaging of Robot Framework so I'll want for that.
